akka-cluster/src/main/scala/akka/cluster/protobuf/ClusterMessageSerializer.scala

```diff
@@ -18,72 +18,65 @@ import scala.collection.JavaConverters._
 import scala.concurrent.duration.Deadline
 import java.io.NotSerializableException
 
+import akka.annotation.InternalApi
 import akka.cluster.InternalClusterAction._
 import akka.cluster.routing.{ ClusterRouterPool, ClusterRouterPoolSettings }
 import akka.routing.Pool
 import com.typesafe.config.{ Config, ConfigFactory, ConfigRenderOptions }
 
 /**
- * Protobuf serializer of cluster messages.
+ * INTERNAL API
  */
-class ClusterMessageSerializer(val system: ExtendedActorSystem) extends BaseSerializer {
+@InternalApi
+private[akka] object ClusterMessageSerializer {
+  // FIXME use short manifests when we can break wire compatibility
+  // needs to be full class names for backwards compatibility
+  val JoinManifest = classOf[InternalClusterAction.Join].getName
+  val WelcomeManifest = classOf[InternalClusterAction.Welcome].getName
+  val LeaveManifest = classOf[ClusterUserAction.Leave].getName
+  val DownManifest = classOf[ClusterUserAction.Down].getName
+  // #24622 wire compatibility
+  val InitJoinManifest = "akka.cluster.InternalClusterAction$InitJoin$" // use this to be able to roll up from 2.5.9
```

Comment:
This is the first half of the fix - `InitJoin` was a singleton but became a case class, we need to accept the old class-name/manifest and also send using that